std/regex/internal/thompson.d

```diff
@@ -914,14 +908,13 @@ struct ThreadList(DataIndex)
     //dispose list of threads
     void recycle(ref ThreadList!DataIndex list)
     {
-        auto t = list.tip;
-        while(t)
+        if(list.tip)
         {
-            auto next = t.next;
-            recycle(t);
-            t = next;
+            // just put this head-tail list in front of freelist
+            list.toe.next = freelist;
+            freelist = list.tip;
+            list = list.init;
         }
-        list = list.init;
```

Comment:
Any chance you could rename tip and toe to head and tail, I've never seen those names in the context of lists.